### ISSUE TITLE
val_loss not making sense with multiple outputs and weighted loss

### ISSUE THREAD

**Issue by xiankgx** [keras-team/keras#13389]
**System information**  
- Have I written custom code (as opposed to using example directory):  Yes, custom Keras Sequence
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Ubuntu 16.04
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.14.0
- Keras version:  2.3.0
- Python version:  Python 3.6.8
- CUDA/cuDNN version:  10.1
- GPU model and memory:  ASUS Dual GeForce® RTX 2080 Ti  (2 pieces)

You can obtain the TensorFlow version with:  
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"  
You can obtain the Keras version with:  
python -c 'import keras as k; print(k.__version__)'  

**Describe the current behavior**  
I have a model with multiple outputs, and hence there is loss for each of the output and an overall weighted loss. The weighted sum of all the validation loss of the different outputs doesn't add up to the overall validation loss, val_loss.

**Describe the expected behavior**  
Expecting val_loss to be a weighted sum of all the validation loss of model output.

**Code to reproduce the issue**  
```
input_score = Input(shape=(None, None, 3), name="regression_input")
input_class = Input(shape=(None, None, 3), name="classification_input")

model_base = ResNet50(weights="imagenet", include_top=False)
gap = GlobalAveragePooling2D()

shared_dense_1 = Dense(2048)
shared_bn_1 = BatchNormalization()
shared_relu_1 = ReLU()

shared_dense_2 = Dense(2048)
shared_bn_2 = BatchNormalization()
shared_relu_2 = ReLU()

x_score = model_base(input_score)
x_score = gap(x_score)

x_score = shared_dense_1(x_score)
x_score = shared_bn_1(x_score)
x_score = shared_relu_1(x_score)

x_score = shared_dense_2(x_score)
x_score = shared_bn_2(x_score)
x_score = shared_relu_2(x_score)

score_out = Dense(1, activation="sigmoid", name="regression_output")(x_score)

x_class = model_base(input_class)
x_class = gap(x_class)

x_class = shared_dense_1(x_class)
x_class = shared_bn_1(x_class)
x_class = shared_relu_1(x_class)

x_class = shared_dense_2(x_class)
x_class = shared_bn_2(x_class)
x_class = shared_relu_2(x_class)
    
class_out = Dense(1, activation="sigmoid", name="classification_output")(x_class)

with tf.device('/cpu:0'):
    model = Model([input_score, input_class], [score_out, class_out])

try:
    parallel_model = multi_gpu_model(model, gpus=gpus)
except:
    parallel_model = model

parallel_model.compile(optimizer=Adam(lr=0.0001), 
              loss=[
                  "mse", 
                  "binary_crossentropy"
              ],
              loss_weights=[
                  20, 
                  1
              ],
              metrics={
                "regression_output": ["mse", "mae"],
                "classification_output": "acc"
              })
parallel_model.summary()

hist = parallel_model.fit_generator(dataset_train, 
                    validation_data=dataset_test,
                    callbacks=callbacks,
                    workers=workers, 
                    shuffle=True, 
                    epochs=10,
                    initial_epoch=0)
```

**Other info / logs**  
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by xiankgx** [comment]
Here are the 3 columns of data: val_loss, val_regression_output_loss, and val_classification_output_loss with weights [20, 1], 20 for val_regression_output_loss and 1 for val_classification_output_loss.

val_loss:
0.169534176588059
0.407849103212357
0.244819268584251
0.149829715490341
0.144160956144333
0.200807094573975
0.074971824884415
0.100794419646263
0.071464695036411
0.384763598442078

val_regression_output_loss:
0.004071841016412
0.004012452904135
0.004082964267582
0.004099677316844
0.004057640675455
0.004182121250778
0.004099731799215
0.004059502854943
0.004109856206924
0.004216863773763

val_classification_output_loss:
0.135637700557709
0.145183801651001
0.136177375912666
0.146444052457809
0.128644600510597
0.173463732004166
0.132748082280159
0.152826607227325
0.160107001662254
0.138926386833191

**Comment by xiankgx** [comment]
I replaced the parralel_model with conventional single GPU model and get the same irratic val_loss too. What could be the problem?

**Comment by xiankgx** [comment]
All this is run inside a docker container based on a custom Docker image based on 
tensorflow/tensorflow:latest-gpu-py3-jupyter

FROM tensorflow/tensorflow:latest-gpu-py3-jupyter

```
RUN apt-get update
RUN apt-get install --yes nano vim
RUN apt-get install --yes python3-pip
# RUN apt-get install --yes nodejs npm
RUN apt-get install --yes libsm6 libxext6 libxrender-dev
# RUN apt-get install --yes git

# RUN npm install -g configurable-http-proxy

# RUN pip3 install jupyterhub
RUN pip3 install notebook
RUN pip3 install ipython
RUN pip3 install scipy
RUN pip3 install pandas
RUN pip3 install "dask[complete]"
RUN pip3 install seaborn
RUN pip3 install plotly
RUN pip3 install scikit-learn
RUN pip3 install xgboost
RUN pip3 install lightgbm
RUN pip3 install catboost
RUN pip3 install imbalanced-learn
RUN pip3 install keras
RUN pip3 install torch==1.0 torchvision
RUN pip3 install opencv-python
RUN pip3 install scikit-image
RUN pip3 install imgaug
#RUN pip3 install statsmodel
RUN pip3 install nltk
RUN pip3 install gensim

CMD ["/bin/bash"]

```

**Comment by xiankgx** [comment]
Training loss is ok, but val_loss don't add up. This is weirld...

**Comment by xiankgx** [comment]
Output of lsb_release -a

```
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.6 LTS
Release:        16.04
Codename:       xenial

```

**Comment by bfialkoff** [comment]
I am experiencing the same issue, with tf 1.13.1, 1.14.0, and 2.0. Also for me the training loss is fine but the val_loss doesnt add up:
`loss: 0.3240 - regression_loss: 0.2500 - classification_loss: 0.0740 - val_loss: 8.6638e-07 - val_regression_loss: 0.3861 - val_classification_loss: 0.2899
`

**Comment by xiankgx** [comment]
This is giving me lots of confusion in my work projects. I just tried with replacing keras with tf.keras and it seems to work as expected. I guess Keras is being abandoned.

**Comment by theoVag** [comment]
Hello,
I have the exact problem but I didn't understand how you solve it. Could you explain me?
Thankss in advance

**Comment by xiankgx** [comment]
> Hello,
> I have the exact problem but I didn't understand how you solve it. Could you explain me?
> Thankss in advance

Try different combinations of TF + Keras, maybe TF 1.13.1 and Keras 2.2.4.
Or use tf.keras instead of keras.

**Comment by theoVag** [comment]
Thanks for the fast response, the problem solved when I downgraded the keras to 2.2.5

**Comment by gagolucasm** [comment]
Same problem here, solved by moving to tensorflow keras

**Comment by Gabry993** [comment]
I have the same problem. I've noticed that when passing a fixed validation set (as array) the losses sum up correctly. When using a generator for the validation (implemented with keras sequence), the overall loss doesn't make sense anymore (while it makes sense for the training set, implemented with the same generator)
